sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/main/java/com/azure/data/schemaregistry/apacheavro/SchemaRegistrySchemaCache.java

```diff
@@ -0,0 +1,222 @@
+     */
+    int getSize() {
+        synchronized (lock) {
+            return cache.size();
+        }
+    }
+
+    /**
+     * Gets the length of schemas stored in cache.
+     *
+     * @return The length of schemas stored in cache.
+     */
+    int getTotalLength() {
+        synchronized (lock) {
+            return cache.getTotalLength();
+        }
+    }
+
+    /**
+     * Logs the cache status if log level verbose is enabled. Otherwise, no-op.
+     */
+    private void logCacheStatus() {
+        if (!logger.canLogAtLevel(LogLevel.VERBOSE)) {
+            return;
+        }
+
+        final int size = cache.size();
+        final int length = cache.getTotalLength();
+
+        logger.atVerbose()
+            .addKeyValue(SIZE_KEY, size)
+            .addKeyValue(TOTAL_LENGTH_KEY, length)
+            .log("Cache entry added or updated. Total number of entries: {}; Total schema length: {}",
+                size, length);
+    }
+
+    /**
+     * Simple LRU cache. Accesses to cache are synchronized via the outer class lock.
+     */
+    private static final class SchemaCache extends LinkedHashMap<String, Schema> {
+        private static final long serialVersionUID = -1L;
+
+        private final int capacity;
+        private final HashMap<Schema, String> schemaToIdCache = new HashMap<>();
+
+        private int totalLength;
+
+        /**
+         * Creates an LRU cache with maximum capacity.
+         *
+         * @param capacity Max size (number of entries) of the cache.
+         */
+        SchemaCache(int capacity) {
+            super(64, 0.75f, true);
+            this.capacity = capacity;
+        }
+
+        int getTotalLength() {
+            return totalLength;
+        }
+
+        String getSchemaId(Schema schema) {
+            final String schemaId = schemaToIdCache.get(schema);
+
+            if (schemaId != null) {
+                // Simulate an access so that the entry does not expire.
+                super.get(schemaId);
+            }
+
+            return schemaId;
+        }
+
+        /**
+         * Adds a schema keyed by its schema id.  If it is larger than its counterpart.
+         */
+        @Override
+        public Schema put(String schemaId, Schema value) {
+            final Schema existing = super.put(schemaId, value);
+            final int currentLength = value.toString().length();
+
+            // The replaced node may be of a different size.
+            if (existing == null) {
+                totalLength = totalLength + currentLength;
+            } else {
+                final int difference = currentLength - existing.toString().length();
+                totalLength = totalLength - difference;
+            }
+
+            schemaToIdCache.put(value, schemaId);
+
+            return existing;
+        }
+
+        @Override
+        protected boolean removeEldestEntry(Map.Entry<String, Schema> eldest) {
+            final boolean removingEntry = size() > capacity;
```

Comment:
If we evict an entry from the cache, then the entry we removed should decrement from the totalLength, since its schema is no longer in the map.

Reply:
Yeah, I think you answered my question in the previous comment, I was looking at why totalLength was needed if the eviction policy is based on count of entries and not totalLength.

Reply:
Oh yeah... They wanted the totalLength to estimate "how big is this cache?" since a Schema can be fairly large and may be a single entry in the cache.